### PR TITLE
Custom profiling in kubectl debug feature blog

### PR DESCRIPTION
**Pull request by ardaguclu**
Feature blog ~place holder~ article for https://github.com/kubernetes/enhancements/issues/4292

### CODE REVIEW

**Comment by sftim** [comment]
/hold

pending assignment of publication date (by release comms)

we'll also keep this held until the v1.31 release has happened

**Comment by Princesso** [comment]
Hello @ardaguclu :wave: please take a look at [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to get your PR ready for review before Tuesday July 16th 2024 18:00 PST. Thank you!

**Comment by ardaguclu** [comment]
This is ready for review.
/cc @soltysh @eddiezane

File: content/en/blog/_posts/2024-06-21-custom-profiling-kubectl-debug.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    - name: REQUIRED_ENV_VAR
+      value: "value1"
+```
+
+Currently, copying the pod is the sole mechanism that supports debugging this pod in kubectl debug. Furthermore, what if user needs to modify the `REQUIRED_ENV_VAR` to something different
+for advanced troubleshooting?. There is no mechanism to achieve this.
+
+## Custom Profiling
+
+Custom profiling is a new flag, namely `custom`, introduced in kubectl debug to provide extensibility. It expects partial `Container` spec in either YAML or JSON format.
```
Comment (soltysh) [inline content/en/blog/_posts/2024-06-21-custom-profiling-kubectl-debug.md:43]:
```suggestion
Custom profiling is a new functionality available under `--custom` flag, introduced in kubectl debug to provide extensibility. It expects partial `Container` spec in either YAML or JSON format. 
```

**Review by soltysh — APPROVED** [review]
Minor nits, but lgtm from sig-cli pov

File: content/en/blog/_posts/2024-06-21-custom-profiling-kubectl-debug.md (hunk 1 above)
Comment (ardaguclu) [reply content/en/blog/_posts/2024-06-21-custom-profiling-kubectl-debug.md]: Thank you, committed.

**Comment by edithturn** [comment]
Hello @ardaguclu 🙂 , it's Edith Puclla from the 1.31 Release comms team.

A friendly reminder that the deadline for the blog to be ready for review is July 26, 2024.

Let me know if there's anything we can do to help with content, review, or anything else! 🙂

**Comment by ardaguclu** [comment]
> Hello @ardaguclu 🙂 , it's Edith Puclla from the 1.31 Release comms team.
> 
> A friendly reminder that the deadline for the blog to be ready for review is July 26, 2024.
> 
> Let me know if there's anything we can do to help with content, review, or anything else! 🙂

@edithturn thank you for reminding. This PR is ready to merge actually, just waiting from approval from sig-docs.

**Comment by tengqm** [comment]
Contents lgtm.
/lgtm
Pending publish date assignment.

**Comment by edithturn** [comment]
> > Hello @ardaguclu 🙂 , it's Edith Puclla from the 1.31 Release comms team.
> > A friendly reminder that the deadline for the blog to be ready for review is July 26, 2024.
> > Let me know if there's anything we can do to help with content, review, or anything else! 🙂
> 
> @edithturn thank you for reminding. This PR is ready to merge actually, just waiting from approval from sig-docs. According to this comment [#46909 (comment)](https://github.com/kubernetes/website/pull/46909#issuecomment-2185044362), this PR is waiting from you about the publication date?.

Hello @ardaguclu, Thank you so much for your update. 
This blog's publish date is **August 22nd.**

**Comment by ardaguclu** [comment]
@edithturn thank you, I've updated the PR withdetermined publication date.

**Comment by ardaguclu** [comment]
@tengqm could you please have a look at this PR?. Thank you.

**Comment by sftim** [comment]
/hold

see https://github.com/kubernetes/website/pull/46909#issuecomment-2252545970

**Comment by ardaguclu** [comment]
@sftim we have updated the publication date, see https://github.com/kubernetes/website/pull/46909#issuecomment-2252330343
Do you still think that we should hold this PR?

**Comment by sftim** [comment]
>  Do you still think that we should hold this PR?

Yes, because it's like a logical AND; the other condition (has v1.31 been released yet) is not met.

**Comment by ardaguclu** [comment]
> > Do you still think that we should hold this PR?
> 
> Yes, because it's like a logical AND; the other condition (has v1.31 been released yet) is not met.

Thank you. You are right.

**Review by sftim — COMMENTED** [review]
Some feedback

**Comment by ardaguclu** [comment]
@sftim thank you for your review. I've incorporated the changes you suggested. Could you PTAL?. Thank you.

**Review by divya-mohan0209 — COMMENTED** [review]
Thanks for this!
/approve

**Comment by sftim** [comment]
I'd been hoping another contributor might LGTM this,  but as that's not happened:
/lgtm
/hold cancel

**Pull request merged**